### PR TITLE
Issue/36/deposit

### LINKED ISSUE #36: Add test for DepositContract and fix impl

### PR DESCRIPTION
**Pull request by syuhei176**
Add test for deposit contract

Close #36

### CODE REVIEW

File: contracts/DepositContract.sol [hunk 1]
```diff
@@ -43,13 +43,12 @@ contract DepostiAndExit {
         erc20.transferFrom(msg.sender, address(this), _amount);
         types.Range memory depositRange = types.Range({start:totalDeposited, end:totalDeposited + _amount});
         types.StateUpdate memory stateUpdate = types.StateUpdate({
-            property:_initialState,
+            stateObject: _initialState,
             range: depositRange,
             plasmaBlockNumber: getLatestPlasmaBlockNumber(),
```
Comment (syuhei176) [inline contracts/DepositContract.sol]: I'm thinking how to get and define deposit block number

File: contracts/DepositContract.sol [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    event LogCheckpoint(
+        types.Checkpoint checkpoint
+    );
+
+    event ExitFinalized(
+        bytes32 exitId
+    );
+
+    /* Public Variables and Mappings*/
+    ERC20 public erc20;
```
Comment (shogochiai) [inline contracts/DepositContract.sol]: How is the timing of making an abstract Deposit contract to switch ERC-20 and ERC-721?

Comment (syuhei176) [reply contracts/DepositContract.sol]: I have no idea about timing now. but we should create another DepositContract for ERC721.

Comment (syuhei176) [reply contracts/DepositContract.sol]: ah we can create DepositContract standard like
https://github.com/cryptoeconomicslab/plasma-predicates/blob/master/contracts/standard/DepositStandard.sol. maybe after Dec 2019.

Comment (shogochiai) [reply contracts/DepositContract.sol]: Cool💪

Comment (syuhei176) [reply contracts/DepositContract.sol]: thanks! also you can write code for ERC721.

**Pull request merged**